tests/test_templates.py

```diff
@@ -12,106 +12,307 @@
-            return qml.expval(qml.Identity(0))
+    @pytest.mark.parametrize(fixture, qubit_const)
+    def test_integration_qubit_keyword_arg(self, template, inpts):
+        """Checks integration of qubit templates using keyword arguments to qnode."""
 
-        circuit(weights=p)
+        inpts = [np.array(i) for i in inpts]
+        dev = qml.device('default.qubit', wires=2)
 
-    @pytest.mark.parametrize('parfun', [random_layers_uniform, random_layers_normal])
-    def test_integration_random_layers(self, parfun, qubit_device, n_subsystems, n_layers):
-        """Checks that parameters generated by methods from pennylane.init integrate
-        with pennnylane.templates.layers.RandomLayers()."""
+        @qml.qnode(dev)
+        def circuit(inp_=None):
+            qml.PauliX(wires=0)
+            template(*inp_, wires=range(2))
+            template(*inp_, wires=range(2))
+            qml.PauliX(wires=1)
+            return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
 
-        p = parfun(n_layers=n_layers, n_wires=n_subsystems)
+        circuit(inp_=inpts)
 
-        @qml.qnode(qubit_device)
-        def circuit(weights):
-            RandomLayers(*weights, wires=range(n_subsystems))
-            return qml.expval(qml.Identity(0))
+    @pytest.mark.parametrize(fixture, cv_const)
+    def test_integration_cv_positional_args(self, gaussian_device_2_wires, template, inpts):
+        """Checks integration of continuous-variable templates using positional arguments to qnode."""
 
-        circuit(weights=p)
+        inpts = [np.array(i) for i in inpts]
 
-    @pytest.mark.parametrize('parfun', [random_layer_uniform, random_layer_normal])
-    def test_integration_random_layer(self, parfun, qubit_device, n_subsystems):
-        """Checks that parameters generated by methods from pennylane.init integrate
-        with pennnylane.templates.layers.RandomLayer()."""
+        @qml.qnode(gaussian_device_2_wires)
+        def circuit(inp_):
+            qml.Displacement(1., 1., wires=0)
+            template(*inp_, wires=range(2))
+            template(*inp_, wires=range(2))
+            qml.Displacement(1., 1., wires=1)
+            return [qml.expval(qml.Identity(0)), qml.expval(qml.X(1))]
 
-        p = parfun(n_wires=n_subsystems)
+        circuit(inpts)
 
-        @qml.qnode(qubit_device)
-        def circuit(weights):
-            RandomLayer(*weights, wires=range(n_subsystems))
-            return qml.expval(qml.Identity(0))
+    @pytest.mark.parametrize(fixture, cv_const)
+    def test_integration_cv_keyword_args(self, gaussian_device_2_wires, template, inpts):
+        """Checks integration of continuous-variable templates using keyword arguments to qnode."""
+
+        inpts = {"w"+str(i): np.array(inpts[i]) for i in range(len(inpts))}
+
+        @qml.qnode(gaussian_device_2_wires)
+        def circuit(**inp_):
+            qml.Displacement(1., 1., wires=0)
+            template(*inp_.values(), wires=range(2))
```

Comment:
any issues foreseen here with dictionary ordering? @josh146?

Reply:
Might be potentially safer/future-proof to explicitly sort values based on keys

Reply:
Good idea!